platform/v06/platform.go

```diff
@@ -1,13 +1,22 @@
 package v06
 
+import (
+	"github.com/buildpacks/lifecycle/api"
+	"github.com/buildpacks/lifecycle/platform/common"
+)
+
 type Platform struct {
-	api string
+	api *api.Version
+	common.Platform
```

Comment:
```suggestion
	previousPlatform common.Platform
```

Reply:
I still somewhat dislike the embedded struct. It's very convenient to embed it, but I feel that it might be better to be more explicit. With the suggestion above, for every method in the interface we'd need to write something like
```
func (p *Platform) SupportsAssetPackages() bool {
	return previousPlatform.SupportsAssetPackages()
}
```
or a new implementation. It is slightly annoying but it might help us not forget to override the previous logic when we need to. And it might make it more obvious to someone new to the code where the implementation is coming from.

That said, I'd be happy to hear (and learn from) other opinions!

Reply:
I don't have a strong opinion here, I lean a bit toward the embedding as it is. Mostly just because of duplication avoided.

Reply:
Hmm... normally I'm not a fan of duplication but I like that you could read the code for any of the platform/vX packages and it would essentially describe the platform spec (granted, we haven't moved all platform-related logic into this package, but eventually...). Having the code mirror the spec seems like a win from an organizational perspective.

One other reason not to embed the struct - take the definition of `API()`:
```
func (p *Platform) API() string {
	return p.api.String()
}
```

If we forgot to add this method to a new platform package, the code would compile but the new platform would return the previous platform's api version. With the `previousPlatform` field, if we forget to add a method the code would not compile. This seems preferable.

Since I inherited this PR I added a7cdb8e with my preferred implementation 😛  but I'm open to changing it back depending on opinions.